sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -581,7 +659,9 @@ async def serialize_property(
 
     # Ensure that we have a value that Protobuf understands.
     if not isLegalProtobufValue(value):
-        raise ValueError(f"unexpected input of type {type(value).__name__}")
+        raise ValueError(
+            f"unexpected input of type {type(value).__name__} for {property_key} in {type(resource_obj).__name__}"
```

Comment:
property_key and resource_obj might be None. This error needs to handle that otherwise we'll get "for None in None" which doesn't read great.

Reply:
Done! Apologies, I missed this
> Yes, some places call serialize_property and we're not in a resource and/or don't have key. In which case the raise ValueError on line 587 need to handle all those cases [obj&key, just obj, just key, neither].